packages/thirdweb/src/wallets/smart/lib/userop.ts

```diff
@@ -49,6 +49,21 @@
 import { getPaymasterAndData } from "./paymaster.js";
 import { generateRandomUint192 } from "./utils.js";
 
+const isDeploying: Map<string, boolean> = new Map();
+
+export const setIsAccountDeploying = (
+  accountContract: ThirdwebContract,
+  value: boolean,
+) => {
+  const key = `${accountContract.address}-${accountContract.chain.id}`;
+  isDeploying.set(key, value);
+};
+
+export const getIsAccountDeploying = (accountContract: ThirdwebContract) => {
+  const key = `${accountContract.address}-${accountContract.chain.id}`;
+  return isDeploying.get(key) || false;
+};
```

Comment:
since all u care about is if something is in a list I'd use a `Set` here instead

something along the lines of

```suggestion
const isDeploying: Set<string> = new Set();

export const setIsAccountDeploying = (
  accountContract: ThirdwebContract,
  value: boolean,
) => {
  const key = `${accountContract.address}-${accountContract.chain.id}`;
  if (value) {
    isDeploying.add(key);
  } else {
    isDeploying.delete(key);
  }
};

export const getIsAccountDeploying = (accountContract: ThirdwebContract) => {
  const key = `${accountContract.address}-${accountContract.chain.id}`;
  return isDeploying.has(key);
};
```

Reply:
also just generally we do `chainid:address` though that part is a complete nitpick on my part here

Reply:
also also: I feel like it would be more clear to have:

`isAccountDeploying(account)`
`markAccountAsDeploying(account)`
`clearAccountDeploying(account)`

(or similar names)

instead of passing the bool

Reply:
done